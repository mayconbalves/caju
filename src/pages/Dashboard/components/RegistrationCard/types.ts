export type Registrations = {
  id: string
  admissionDate: string
  email: string
  employeeName: string
  status: string | any
  cpf: string
}

export type RegistrationsCardProps = {
  registration: Registrations
  onDelete: () => void
  onUpdate: (registration: any, status: string) => void
}
