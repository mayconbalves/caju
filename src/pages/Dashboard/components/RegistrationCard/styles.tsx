import styled from 'styled-components'

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  gap: 15px;
  border: 4px solid #fff;
  margin: 16px;
  border-radius: 8px;
  padding: 16px;
  background-color: #fff;
  border: 1px solid #000;
  box-shadow:
    0 1px 3px rgba(0, 0, 0, 0.12),
    0 1px 2px rgba(0, 0, 0, 0.24);

  &:hover {
    box-shadow:
      0 3px 9px rgba(0, 0, 0, 0.12),
      0 3px 6px rgba(0, 0, 0, 0.24);
  }
`

export const TitleCard = styled.h3`
  margin: 0;
`

export const DataCard = styled.p`
  margin: 0;
`

export const IconAndText = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
`

export const Actions = styled.div`
  margin-top: 8px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 4px;

  svg {
    cursor: pointer;
  }
`
