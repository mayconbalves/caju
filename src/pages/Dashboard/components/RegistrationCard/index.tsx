import { HiOutlineCalendar, HiOutlineMail, HiOutlineTrash, HiOutlineUser } from 'react-icons/hi'
import { ButtonSmall } from '~/components/Buttons'
import * as S from './styles'

type Props = {
  data: any
  onDelete: () => void
  onUpdate: (registration: any, status: string) => void
}

const RegistrationCard = (props: Props) => {
  return (
    <S.Card>
      <S.IconAndText>
        <HiOutlineUser />
        <h3>{props.data.employeeName}</h3>
      </S.IconAndText>
      <S.IconAndText>
        <HiOutlineMail />
        <p>{props.data.email}</p>
      </S.IconAndText>
      <S.IconAndText>
        <HiOutlineCalendar />
        <span>{props.data.admissionDate}</span>
      </S.IconAndText>
      <S.Actions>
        {props.data.status === 'REVIEW' && (
          <>
            <ButtonSmall
              bgcolor="rgb(255, 145, 154)"
              onClick={() => props.onUpdate(props.data, 'REPROVED')}
            >
              Reprovar
            </ButtonSmall>
            <ButtonSmall
              bgcolor="rgb(155, 229, 155)"
              onClick={() => props.onUpdate(props.data, 'APPROVED')}
            >
              Aprovar
            </ButtonSmall>
          </>
        )}

        {['REPROVED', 'APPROVED'].includes(props.data.status) && (
          <ButtonSmall bgcolor="#ff8858" onClick={() => props.onUpdate(props.data, 'REVIEW')}>
            Revisar novamente
          </ButtonSmall>
        )}

        <HiOutlineTrash onClick={props.onDelete} />
      </S.Actions>
    </S.Card>
  )
}

export default RegistrationCard
