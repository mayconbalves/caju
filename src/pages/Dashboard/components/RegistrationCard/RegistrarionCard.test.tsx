import { render, screen } from '@testing-library/react'
import RegistrationCard from '.'

describe('RegistrationCard Component', () => {
  const mockOnDelete = jest.fn()
  const mockOnUpdate = jest.fn()

  const registration = {
    id: '1',
    employeeName: 'John Doe',
    email: 'john@example.com',
    admissionDate: '01/01/2022',
    cpf: '56642105087',
    status: 'REVIEW'
  }

  it('should display the correct registration details', () => {
    render(
      <RegistrationCard
        registration={registration}
        onDelete={mockOnDelete}
        onUpdate={mockOnUpdate}
      />
    )

    expect(screen.getByText('John Doe')).toBeInTheDocument()
    expect(screen.getByText('john@example.com')).toBeInTheDocument()
    expect(screen.getByText('01/01/2022')).toBeInTheDocument()
  })

  it('should render action buttons correctly', () => {
    render(
      <RegistrationCard
        registration={registration}
        onDelete={mockOnDelete}
        onUpdate={mockOnUpdate}
      />
    )

    expect(screen.getByText('Apagar')).toBeInTheDocument()
    expect(screen.getByText('Aprovar')).toBeInTheDocument()
    expect(screen.getByText('Reprovar')).toBeInTheDocument()
  })

  it('should render Approve and Reject buttons when registration is in REVIEW status', () => {
    render(
      <RegistrationCard
        registration={registration}
        onDelete={mockOnDelete}
        onUpdate={mockOnUpdate}
      />
    )

    const approveButton = screen.getByText('Aprovar')
    const rejectButton = screen.getByText('Reprovar')

    expect(approveButton).toBeInTheDocument()
    expect(rejectButton).toBeInTheDocument()
  })

  it('should render Review button when registration is in APPROVED status', () => {
    const registration = {
      id: '1',
      employeeName: 'John Doe',
      email: 'john@example.com',
      admissionDate: '01/01/2022',
      cpf: '56642105087',
      status: 'APPROVED'
    }
    render(
      <RegistrationCard
        registration={registration}
        onDelete={mockOnDelete}
        onUpdate={mockOnUpdate}
      />
    )

    const reviewButton = screen.getByText('Revisar novamente')
    expect(reviewButton).toBeInTheDocument()
  })

  it('should render Review button when registration is in REPROVED status', () => {
    const registration = {
      id: '1',
      employeeName: 'John Doe',
      email: 'john@example.com',
      admissionDate: '01/01/2022',
      cpf: '56642105087',
      status: 'REPROVED'
    }

    const { container } = render(
      <RegistrationCard
        registration={registration}
        onDelete={mockOnDelete}
        onUpdate={mockOnUpdate}
      />
    )

    const reviewButton = screen.getByText('Revisar novamente')
    expect(reviewButton).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
