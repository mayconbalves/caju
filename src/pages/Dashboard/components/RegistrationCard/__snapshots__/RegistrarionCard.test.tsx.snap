// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationCard Component should render Review button when registration is in REPROVED status 1`] = `
<div>
  <div
    class="sc-imWZod liahti"
  >
    <div
      class="sc-kpDprT jplfHT"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <h3
        class="sc-jXbVAB bkqFXX"
      >
        John Doe
      </h3>
    </div>
    <div
      class="sc-kpDprT jplfHT"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <p
        class="sc-dhKdPU inPPyg"
      >
        john@example.com
      </p>
    </div>
    <div
      class="sc-kpDprT jplfHT"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <p
        class="sc-dhKdPU inPPyg"
      >
        01/01/2022
      </p>
    </div>
    <div
      class="sc-kpDprT jplfHT"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4 4a2 2 0 012-2h4.586A2 2 0 0112 2.586L15.414 6A2 2 0 0116 7.414V16a2 2 0 01-2 2H6a2 2 0 01-2-2V4zm2 6a1 1 0 011-1h6a1 1 0 110 2H7a1 1 0 01-1-1zm1 3a1 1 0 100 2h6a1 1 0 100-2H7z"
          fill-rule="evenodd"
        />
      </svg>
      <p
        class="sc-dhKdPU inPPyg"
      >
        56642105087
      </p>
    </div>
    <div
      class="sc-dAlxHm evHRGf"
    >
      <button
        bgcolor="#ff8858"
        class="sc-aYaIB eYtHmf"
        role="button"
      >
        Revisar novamente
      </button>
      <button
        bgcolor="#fc2a2a"
        class="sc-aYaIB eQVzbP"
        role="button"
      >
        Apagar
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9 2a1 1 0 00-.894.553L7.382 4H4a1 1 0 000 2v10a2 2 0 002 2h8a2 2 0 002-2V6a1 1 0 100-2h-3.382l-.724-1.447A1 1 0 0011 2H9zM7 8a1 1 0 012 0v6a1 1 0 11-2 0V8zm5-1a1 1 0 00-1 1v6a1 1 0 102 0V8a1 1 0 00-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
